lib/utils/replace-info.js

```diff
@@ -1,4 +1,4 @@
-const { cleanUrl } = require('npm-registry-fetch')
+const cleanUrl = require('npm-registry-fetch/lib/clean-url')
```

Comment:
We tend not to do this as it unnecessarily couples to the layout of the code, and not its exports.  `module.exports` *is* the module, as far a our semver conventions are concerned.  This allows us to completely refactor or rewrite a module and not make it a breaking change, as long as the exports are the same.

Reply:
The layout of internal files is already part of the api and subject to semver, unless the exports field is used to encapsulate them.

Reply:
so that the future refactor does not introduce destructive changes
I suggest adding an export field to the package.json of this module, will this solve this problem?

Reply:
Something like that would have to be a larger decision/discussion than just one repo, it falls under how we build all of our software.  Currently we define a `main` entry .e.g. https://github.com/npm/npm-registry-fetch/blob/e58e8bc6c5b25d53a764f58190fc3a5c764a2e78/package.json#L5, and rely on `module.exports` to define the api.

`exports` is an alternative to `main`, not a replacement.  We are not moving to `exports` right now so my original, single-line suggestion is all we really want to do right now.

The line as it is now was done very intentionally, and [adding that function to `module.exports`](https://github.com/npm/npm-registry-fetch/pull/107) was the way in which it was exported.

Reply:
To clarify, you suggest leaving `require('npm-registry-fetch')` even though it adds ~15% to the runtime

If the only problem is that you are just starting to migrate to declaring `exports` in `package.json`, then I could first send a PR to `npm-registry-fetch` so that I can painlessly import only `cleanUrl`

Reply:
Yes leave it.  A pr to `npm-registry-fetch` would be rejected as we would want to centrally manage that entry in package.json from `@npmcli/template-oss` and enforce it on ALL our packages.

Reply:
38 milliseconds (15% of 256) for `npm run` is not something we need to hyper optimize.  This is not a hot path like reification.